validphys2/src/validphys/asy_exponents.py

```diff
@@ -0,0 +1,266 @@
+# -*- coding: utf-8 -*-
+"""
+Tools for computing and plotting effective exponents.
+"""
+from __future__ import generator_stop
+
+import logging
+import numbers
+import random
+import warnings
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+from reportengine import collect
+from reportengine.compat import yaml
+from reportengine.figure import figuregen
+from reportengine.floatformatting import format_number, significant_digits, format_error_value_columns
+from reportengine.table import table
+
+from validphys.checks import check_positive, check_pdf_normalize_to, make_argcheck, check_xlimits
+from validphys.core import PDF, FitSpec
+from validphys.pdfbases import check_basis, Basis
+from validphys.pdfplots import BandPDFPlotter, PDFPlotter
+
+import validphys.pdfgrids as pdfgrids
+
+from findiff import FinDiff
```

Comment:
Is this doing anything non trivial? If it was easy I'd rather have the relevant functionality copied somewhere (e.g. calcutils.py) than adding a dependency.

Reply:
This has now been replaced by` np.gradient`, as suggested in #1605